http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/JsonPayloadIT.java

```diff
@@ -0,0 +1,189 @@
+
+        getApp().given()
+                .contentType(ContentType.JSON)
+                .body(jsonPayload)
+                .when()
+                .post("/football/upload-football-json")
+                .then()
+                .statusCode(BAD_REQUEST_STATUS_CODE)
+                .body(containsString(
+                        "Invalid JSON - JsonProcessingException : Unexpected character ('{' (code 123)): was expecting comma to separate Array entries"));
+    }
+
+    @Test
+    public void sendASingleJSONObject() throws JsonProcessingException {
+        FootballTeam teamX = new FootballTeam(
+                "Single Team",
+                "Pink",
+                3,
+                new Date(1900, 1, 1),
+                "A",
+                Arrays.asList("You", "You"));
+        ObjectMapper objectMapper = new ObjectMapper();
+        String jsonPayload = objectMapper.writeValueAsString(teamX);
+
+        getApp().given()
+                .contentType(ContentType.JSON)
+                .body(jsonPayload)
+                .when()
+                .post("/football/upload-football-json")
+                .then()
+                .statusCode(BAD_REQUEST_STATUS_CODE)
+                .body(containsString(
+                        "JsonMappingException : Cannot deserialize value of type `java.util.HashSet<io.quarkus.ts.http.advanced.reactive.FootballTeam>`"));
+
+    }
+
+    @Test
+    public void sendIncorrectContentType() throws IOException {
+        String jsonPayload = readJSONFile("football-empty-teams.json");
+
+        getApp().given()
+                .contentType(ContentType.TEXT)
+                .body(jsonPayload)
+                .when()
+                .post("/football/upload-football-json")
+                .then()
+                .statusCode(UNSUPPORTED_MEDIA_TYPE_STATUS_CODE);
+
+    }
+
+    @Test
+    public void sendSpecialCharactersJSONPayload() throws Exception {
+        FootballTeam team1 = new FootballTeam(
+                "Føøbåll™ Tëâm",
+                "Rëd&Blüé",
+                3,
+                new Date(1900, 1, 1),
+                "Städiümß©",
+                Arrays.asList("لاعب Ône", "Pläÿér Twø´ñ"));
+
+        FootballTeam team2 = new FootballTeam(
+                "Tëâm 2",
+                "Grëèn&Yëllöw",
+                2,
+                new Date(2010, 1, 15),
+                "New Städiüm©",
+                Arrays.asList("Pläÿér プレイ", "Pläÿér Fòur"));
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        String jsonPayload = objectMapper.writeValueAsString(Arrays.asList(team1, team2));
+
+        getApp().given()
+                .contentType(ContentType.JSON)
+                .body(jsonPayload)
+                .when()
+                .post("/football/upload-football-json")
+                .then()
+                .statusCode(CREATION_STATUS_CODE)
+                .body(containsString("Teams added successfully"));
```

Comment:
None of tests in this test class is testing that response was serialized correctly. You don't test that JSON response is valid, just that response contains some string. Additionally,if tests are sending special chars, you might want to check that they are serialized/deserialized correctly.